alpha/alpha-spring-cloud-starter-consul/src/main/java/org/apache/servicecomb/pack/alpha/server/discovery/consul/AlphaConsulAutoConfiguration.java

```diff
@@ -114,7 +114,11 @@ public void listenInstanceRegisteredEvent(InstanceRegisteredEvent instanceRegist
             NewService newservice =  new NewService();
             newservice.setName(service.getServiceName());
             newservice.setId(service.getServiceId());
-            List<String> tags = consulDiscoveryProperties.getTags();
+            newservice.setAddress(service.getAddress());
+            newservice.setPort(service.getServicePort());
```

Comment:
The first time alpha registered the consul, alpha already used the random port, so you don't need to update it. You can see this json data in the agentServiceRegister method of AgentConsulClient class

```java
  public Response<Void> agentServiceRegister(NewService newService, String token) {
    UrlParameters tokenParam = token != null ? new SingleUrlParameters("token", token) : null;
    String json = GsonFactory.getGson().toJson(newService);
    RawResponse rawResponse = this.rawClient.makePutRequest("/v1/agent/service/register", json, new UrlParameters[]{tokenParam});
    if (rawResponse.getStatusCode() == 200) {
      return new Response((Object)null, rawResponse);
    } else {
      throw new OperationException(rawResponse);
    }
  }
```

If you configure server.port=0 when you start alpha, you can see the following data, 56701 is a randomly generated port

```json
{
	"ID": "servicecomb-alpha-server-0-0-0-0-e388eb111394660d63c2286c06ee78a6",
	"Name": "servicecomb-alpha-server",
	"Tags": ["alpha-server-host\u003d0.0.0.0", "alpha-server-port\u003d0", "secure\u003dfalse"],
	"Address": "192.168.1.102",
	"Port": 56701,
	"Check": {
		"Interval": "10s",
		"HTTP": "http://192.168.1.102:56701/actuator/health",
		"Header": {}
	}
}
```

Reply:
[As i describe my development environment above](https://github.com/apache/servicecomb-pack/pull/454#discussion_r277151707), the port will be updated to '0' after call method 
```java
@EventListener
public void listenInstanceRegisteredEvent(InstanceRegisteredEvent instanceRegisteredEvent)
```

I know when project startup, when set server.port=0, 

- spring will generate a random port like 56701
- then spring use this port registered to consul,
- after service registered, spring will publish a `InstanceRegisteredEvent`  event

and alpha server subscribe the event, re-register service instance with `alpha-server-port` and `alpha-server-host` tags while alphaServerPort is set to zero

I think the problem is `NewService newservice =  new NewService();` **after this line, the default health check host and port will be set to null**

and in actually register code in blow
```java
	@Override
	public Response<Void> agentServiceRegister(NewService newService, String token) {
		UrlParameters tokenParam = token != null ? new SingleUrlParameters("token", token) : null;

		String json = GsonFactory.getGson().toJson(newService);
		RawResponse rawResponse = rawClient.makePutRequest("/v1/agent/service/register", json, tokenParam);

		if (rawResponse.getStatusCode() == 200) {
			return new Response<Void>(null, rawResponse);
		} else {
			throw new OperationException(rawResponse);
		}
	}
```
it only serialize `newService` to json, and **the serialization only include non-null value**, so the health check host and port lost

So re-think about this problem, I think the host and port should be set to `newService` is necessary